goal_src/engine/physics/trajectory.gc

```diff
@@ -5,3 +5,172 @@
+    (let ((a1-2 (-> obj initial-velocity)))
+     (let ((v1-1 arg1)
+           (a0-4 arg0)
+           )
+      (.lvf vf4 (&-> v1-1 quad))
+      (.lvf vf5 (&-> a0-4 quad))
+      )
+     (.mov.vf vf6 vf0 :mask #b1000)
+     (.sub.vf vf6 vf4 vf5 :mask #b111)
+     (.svf (&-> a1-2 quad) vf6)
+     )
+    (vector-xz-normalize! (-> obj initial-velocity) f0-3)
+    )
+   (set!
+    (-> obj initial-velocity y)
+    (- (/ (- (-> arg1 y) (-> arg0 y)) arg2) (* (* 0.5 arg2) (-> obj gravity)))
+    )
+   (none)
+   )
+  )
+
+;; definition for method 12 of type trajectory
+;; INFO: Return type mismatch int vs none.
+(defmethod
+  TODO-RENAME-12
+  trajectory
+  ((obj trajectory) (arg0 vector) (arg1 vector) (arg2 float) (arg3 float))
+  (let ((f0-1 (/ (vector-vector-xz-distance arg1 arg0) arg2)))
+   ((method-of-type trajectory TODO-RENAME-11) obj arg0 arg1 f0-1 arg3)
+   )
+  (none)
+  )
+
+;; definition for method 13 of type trajectory
+;; INFO: Return type mismatch int vs none.
+(defmethod
+  TODO-RENAME-13
+  trajectory
+  ((obj trajectory) (arg0 vector) (arg1 vector) (arg2 float) (arg3 float))
+  (let* ((f0-0 arg2)
+         (f1-3 (- (* f0-0 f0-0) (* (* 2.0 (- (-> arg0 y) (-> arg1 y))) arg3)))
+         (f0-3 900.0)
+         )
+   (when (>= f1-3 0.0)
+    (let ((f0-4 (sqrtf f1-3)))
+     (set! f0-3 (fmax (/ (- (- arg2) f0-4) arg3) (/ (+ (- arg2) f0-4) arg3)))
+     )
+    )
+   ((method-of-type trajectory TODO-RENAME-11) obj arg0 arg1 f0-3 arg3)
+   )
+  (none)
+  )
+
+;; definition for method 14 of type trajectory
+;; INFO: Return type mismatch int vs none.
+(defmethod
+  TODO-RENAME-14
+  trajectory
+  ((obj trajectory) (arg0 vector) (arg1 vector) (arg2 float) (arg3 float))
+  (let* ((f1-2 (+ arg2 (fmax (-> arg0 y) (-> arg1 y))))
+         (f1-5 (* (* 2.0 (- (-> arg0 y) f1-2)) arg3))
+         (f0-3 4096.0)
+         )
+   (if (< 0.0 f1-5)
+    (set! f0-3 (sqrtf f1-5))
+    )
+   ((method-of-type trajectory TODO-RENAME-13) obj arg0 arg1 f0-3 arg3)
+   )
+  (none)
+  )
+
+;; definition for method 15 of type trajectory
+;; INFO: Return type mismatch int vs none.
+;; Used lq/sq
+(defmethod TODO-RENAME-15 trajectory ((obj trajectory))
+  (let ((s5-0 (new 'stack-no-clear 'trajectory))
+        (s4-0 (new 'stack-no-clear 'trajectory))
+        (s3-0 10)
+        )
+   (let ((v1-0 s4-0))
+    (set! (-> v1-0 initial-position quad) (-> obj initial-position quad))
+    )
+   (dotimes (s2-0 s3-0)
+    (let ((v1-1 s5-0))
+     (set! (-> v1-1 initial-position quad) (-> s4-0 initial-position quad))
+     )
+    (let
+     ((f0-1 (* (-> obj time) (/ (+ 1.0 (the float s2-0)) (the float s3-0)))))
+     ((method-of-type trajectory TODO-RENAME-9) obj f0-1 s4-0)
+     )
+    (let ((t9-1 add-debug-line)
+          (a0-6 #t)
+          (a1-1 68)
+          (a2-1 s5-0)
+          (a3-0 s4-0)
+          (t0-0 #x800000ff)
+          (t1-0 #f)
+          )
+     (let ((t2-0 -1))
+      )
+     (t9-1 a0-6 a1-1 a2-1 a3-0 (the-as int t0-0) t1-0)
```

Comment:
based on the code here, i'd guess that `add-debug-line` takes one more argument (possibly unused inside the function).   
If you try that, does this get a lot nicer?